datastream-server/src/main/java/com/linkedin/datastream/server/assignment/LoadBasedPartitionAssigner.java

```diff
@@ -373,39 +360,31 @@ public String toJson() throws IOException {
       return JsonUtils.toJson(this);
     }
   }
-  /**
-   * Encapsulates assignment metrics for a single datastream group
-   */
-  private static class PartitionAssignmentStats {
-    private final int _minPartitionsAcrossTasks;
-    private final int _maxPartitionsAcrossTasks;
 
-    public static final PartitionAssignmentStats DEFAULT = new PartitionAssignmentStats(0, 0);
+  private static class DatastreamMetrics {
```

Comment:
no. I'd think we'd track that in the outer class and provide getters.

I do see that there is an incidental `public` here, so one could argue I'm accidentally dropping a public API. Concern?

Reply:
I think that was one of the idea of the original code to track those values so that they can be exposed using getters. With the new code, that data structure is dropped.

Reply:
Thoughts? We seem to be stuck on this issue. I was trying to minimize the lifecycle mgmt we are doing here, since this class doesn't really have any hooks into when datastreams are created or deleted etc. I think it's a bad idea to piggyback on the metrics registration/deregistration hooks that we do have.

If we need to track these values and expose via getters, I think we'd need to add more lifecycle hooks. I'm skeptical we'd do that in the interface we're impl'ing here.

Reply:
Seems like we're not exposing these outside this class at the moment. I'd suggest to worry about how that is happening if there's really a need to use these metric values outside `LoadBasedPartitionAssigner`